generatorcore/agri2018/energy_demand/energyWithPercentage.py

```diff
@@ -0,0 +1,17 @@
+# pyright: strict
+
+from dataclasses import dataclass
+
+from ...utils import div
+
+from .energy import Energy
+
```

Comment:
I don't feel super strongly but for cases like this I think the split into one file per class is actually net negative.

I mean in this case both when I see the energy class seeing that we also have a EnergyWithPercentage class is good context and the other way around and in both cases everything would still fit on a single screen.

Or differently a module is a unit of abstraction / encapsulation.  We should use it to actually group related things together.  In my point of view I would also put the Co2eFrom* classes into one file (maybe even co2efrom.Soil) .  But the tradeoff is less clear the longer the class.